test/tools/ossfuzz/solc_ossfuzz.cpp

```diff
@@ -38,17 +38,23 @@ extern "C" int LLVMFuzzerTestOneInput(uint8_t const* _data, size_t _size)
 		{
 			TestCaseReader t = TestCaseReader(std::istringstream(input));
 			sourceCode = t.sources().sources;
+			map<string, string> settings = t.settings();
+			bool compileViaYul =
```

Comment:
Just wondering if this can be set to `true` all the time, since we have full Yul coverage now.

Reply:
We have the following cases

- optimise, ir
- no optimise, ir
- optimise, no ir
- no optimise, no ir

If we unconditionally set `m_compileViaYul` to true, we will not be able to catch bugs in the last two configurations i.e., old codegen (plus optimiser) iiuc.

Reply:
Ok. Just to clarify, which tests are it reading, the semantic tests?

Reply:
The fuzzer accepts all tests in `test` subdirectory with the file extension ".sol" as input (so called seeds) and starts mutating them randomly.

Reply:
Doesn't it mean that the syntax tests would be using the old codegen. And most of the semantic tests would be using the new codegen?

Reply:
We have the following buckets as seeds (initial input)

- syntax tests with no `settings` (i.e., `compileViaYul` and/or `optimize`)
- semantic tests with sometimes `compileViaYul` sometimes not, similarly with the `optimize` setting

If there are no mutations whatsoever, you are right that a mapping syntax->old codegen and semantics->old(new)codegen will develop.

However, I expect the fuzzer to occasionally place `// compileViaYul: true` inside syntax tests making them trigger new codegen. Likewise with the `// optimize: true` triggering optimisation on syntax tests.

The tricky part for the fuzzer to reverse engineer is that settings must follow a line with `====` placed at the end of source code. I think this is something that may require a follow up PR. Also, another case that will need care is if fuzzer places two different settings (my guess is the last setting takes precedence since it overwrites the map at a given key).

Reply:
Ok! I was mostly asking, if fuzzing old codegen makes sense, since it would get obsolete sometime in the future (hopefully, soon.) Maybe we can decide that later when it's done. I'm fine with merging this now.

Reply:
That got me thinking that we should facilitate such TestReader mutations. That's why I updated this PR with additions to solidity fuzzing dictionary with tokens that have a special meaning in the test parser (`test/TestCaseReader.cpp`).